go/ephemeral/user_ek_box_storage.go

```diff
@@ -100,17 +101,17 @@ func (s *UserEKBoxStorage) Get(ctx context.Context, generation keybase1.EkGenera
 
 	// Try cache first
 	cacheItem, ok := cache[generation]
-	if ok {
-		defer s.Unlock() // release the lock after we unbox
-		if cacheItem.HasError() {
-			return userEK, cacheItem.Error()
-		}
-		return s.unbox(ctx, generation, cacheItem.UserEKBoxed)
+	if !ok {
+		// We don't have anything in our cache, fetch from the server
+		s.Unlock() // release the lock while we fetch
+		return s.fetchAndStore(ctx, generation)
 	}
 
-	// We don't have anything in our cache, fetch from the server
-	s.Unlock() // release the lock while we fetch
-	return s.fetchAndStore(ctx, generation)
+	defer s.Unlock() // release the lock after we unbox
+	if cacheItem.HasError() {
+		return userEK, cacheItem.Error()
+	}
+	return s.unbox(ctx, generation, cacheItem.UserEKBoxed)
```

Comment:
this didn't change behavior I think

Reply:
yeah i changed this so it matched the team storage more closely